### ISSUE TITLE
ComponentSettings not being cloned when creating new Session

### ISSUE THREAD

**Issue by beepistry** [opentap/opentap#710]
With reference to my forum post [here](https://forum.opentap.io/t/resultsettings-in-new-session-behaviour/774), ```ComponentSettings```  aren't being cloned into a new ```Session```. I have one project where this only happens for the ```ResultSettings```, and the code I created to post here for reproducing the issue, where all ```ComponentSettings``` are affected.

Here's the code to reproduce:
```cs
class Program
{
    static void Main(string[] args)
    {
        ResultSettings.Current.Add(new MyResultListener());
        InstrumentSettings.Current.Add(new MyInstrument());
        DutSettings.Current.Add(new MyDut());

        Console.WriteLine($"From main session: {Session.Current.Id}");
        PrintSettings();

        using (Session.Create(SessionOptions.OverlayComponentSettings | SessionOptions.RedirectLogging))
        {
            Console.WriteLine($"From new session: {Session.Current.Id}");
            PrintSettings();
        }
    }

    static void PrintSettings()
    {
        ResultSettings.Current.Select(rs => rs.Name).ToList().ForEach(s => Console.WriteLine(s));
        InstrumentSettings.Current.Select(rs => rs.Name).ToList().ForEach(s => Console.WriteLine(s));
        DutSettings.Current.Select(rs => rs.Name).ToList().ForEach(s => Console.WriteLine(s));
    }
}

class MyResultListener : ResultListener
{
    public MyResultListener()
    {
        Name = "My Result Listener";
    }
}

class MyInstrument : Instrument
{
    public MyInstrument()
    {
        Name = "My Instrument";
    }
}

class MyDut : Dut
{
    public MyDut()
    {
        Name = "MyDut";
    }
}
```

**Comment by beepistry** [comment]
Thanks, Rolf. I had a feeling when I saw two different behaviours that it was something I was doing. I'm now seeing the ComponentSettings in the fresh Sessions.

Unfortunately, I still can't get the functionality I was after, which was to create some synchronisation between test runs - I tried with the EventResultlistener and a CallbackListener but I guess something in the cloning prevents that from working. Now I'm creating my CallbackListener inside the new session and setting the callbacks from there. It's not as tidy but it should get me where I'm going.
